### PR TITLE
Add feature CategoricalDataCompositor

### PR DESCRIPTION
**Pull request by strandgren**
This PR adds a CategoricalDataCompositor class that allows for the recategorization of categorical data. This is for example useful to combine comparable categories into a common category. The main reason is to be able to load a binary FCI cloud mask derived from the official EUMETSAT FCI cloud mask (CLM) product which contains ten different categories with information about clear sky, meteorological clouds, dust, volcanic ash, scene opacity and missing data. A composite recipe for this has been added to `satpy/etc/composites/fci.yaml`.

A look-up-table (LUT) containing the flags of the new categories must be provided as input when the class in initialized.. The values in the categorical data array that shall be recategorized will then be used as indices to extract the new category flag from the LUT.  Below is an example on how to create a binary clear-sky/cloud  mask from a pseodu cloud type product with six categories representing clear sky (cat1/cat5), cloudy features (cat2-cat4) and missing/undefined data (cat0).

```
cloud_type = local_scene['cloud_type']  # 0 - cat0, 1 - cat1, 2 - cat2, 3 - cat3, 4 - cat4, 5 - cat5,
lut = [np.nan, 0, 1, 1, 1, 0]
compositor = CategoricalDataCompositor('binary_cloud_mask', lut=lut)
composite = compositor([cloud_type])  # 0 - cat1/cat5, 1 - cat2/cat3/cat4, nan - cat0
```

 - [X] Tests added 
 - [X] Fully documented

### CODE REVIEW

File: satpy/composites/__init__.py
Comment (mraspaud) [inline satpy/composites/__init__.py:292]: Triggering a computation here will break the laziness of the processing, are you sure you want to do this?
An alternative for example is to clip the data to the length of the lut: `data = data.clip(max=len(self.lut) - 1)`

File: satpy/composites/__init__.py
Comment (mraspaud) [inline satpy/composites/__init__.py:284]: This mechanism, along with the map_blocks call seems to be used in multiple places now (calibration in the electrol reader, @ameraner 's PR on index maps for FCI). Could you and @ameraner make this a generic utility?

File: doc/source/composites.rst
Comment (mraspaud) [inline doc/source/composites.rst:236]: I'm wondering if we could make the mapping between old and new values clearer for the user? For example like this:

```suggestion
   # categories: 0     1  2  3  4  5
    >>> lut = [np.nan, 0, 1, 1, 1, 0]
```

**Review by mraspaud — COMMENTED** [review]
Good work, clean PR. Some comments inline

File: satpy/composites/__init__.py
Comment (strandgren) [reply satpy/composites/__init__.py]: After some extra thought I simply removed the check and will leave this to be catched by the `IndexError`. I also added to the documentation that the length of `lut` must be greater than the maximum value in `data` in order to avoid an `IndexError`.

File: satpy/composites/__init__.py
Comment (strandgren) [reply satpy/composites/__init__.py:284]: Either me or @ameraner will do a refactoring PR for this once our PRs have been merged.

File: doc/source/composites.rst
Comment (strandgren) [reply doc/source/composites.rst:243]: Good idea, I've added some extra information/explanation on the category remapping.

**Review by pnuu — APPROVED** [review]
LGTM

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**
